posthog/temporal/batch_exports/redshift_batch_export.py

```diff
@@ -153,13 +153,17 @@ async def amerge_identical_tables(
         delete_query = sql.SQL("""\
         DELETE FROM {stage_table}
         USING {final_table} AS final
-        WHERE {merge_condition} AND {stage_table}.{stage_version_key} > final.{final_version_key};
+        WHERE {merge_condition}
+        AND {stage_table}.{stage_person_version_key} < final.{final_person_version_key}
+        AND {stage_table}.{stage_person_distinct_id_version_key} < final.{final_person_distinct_id_version_key};
```

Comment:
`stage_version_key > final_version_key` was wrong as we are deleting from stage the fields that do not need to be merged back into final. So, we should delete those fields for which there is already a higher version present in final.